flocker/common/test/test_retry.py

```diff
@@ -193,6 +194,32 @@ def predicate():
             str(self.failureResultOf(d).value),
             str(LoopExceeded(predicate, False)))
 
+    @capture_logging(None)
+    def test_partial_predicate(self, logger):
+        """
+        Predicate can be a functools.partial function.
+        """
+        result = object()
+
+        def check():
+            return result
+        predicate = partial(check)
+
+        clock = Clock()
+        d = loop_until(clock, predicate)
+        self.assertEqual(
+            self.successResultOf(d),
+            result)
+
+        action = LoggedAction.of_type(logger.messages, LOOP_UNTIL_ACTION)[0]
```

Comment:
:question: You could use list unpacking here instead to ensure that there is only one `LOOP_UNTIL_ACTION`.